File: 2021-July/30/src/member/model/vo/Silver.kt
package member.model.vo

class Silver {

    var name: String? = null
    var grade: String? = null
    var point: Int = 0

    // * https://stackoverflow.com/questions/40352879/what-is-the-equivalent-of-java-static-final-fields-in-kotlin

    // * https://kimch3617.tistory.com/entry/Kotlin%EC%97%90%EC%84%9C-val-%EC%99%80-const-val-%EC%9D%98-%EC%B0%A8%EC%9D%B4

    companion object {
        private const val interest: Float = 0.05f

    }
    var companion = Companion

    // * https://velog.io/@conatuseus/Kotlin-%EC%83%9D%EC%84%B1%EC%9E%90-%EB%BF%8C%EC%8B%9C%EA%B8%B0

    constructor(name: String?, grade : String, point: Int) {
        this.name = name
        this.grade = grade
        this.point = point
    }

    fun getInterest(): Float {
        return this.companion.interest

    }
}

File: 2021-July/30/src/member/model/vo/Silver.cs
namespace member.model.vo
{
    public class Silver
    {
        private string name;
        private string grade;
        private int point;
        private static float interest = 0.02f;

        public Silver() { }

        public Silver(string name, string grade, int point)
        {
            this.name = name;
            this.grade = grade;
            this.point = point;
        }

        public float getInterestPoint()
        {
            return this.point * interest;
        }

        public string getName()
        {
            return this.name;
        }

        public void setName(string name)
        {
            this.name = name;
        }

        public string getGrade()
        {
            return this.grade;
        }

        public void setGrade(string grade)
        {
            this.grade = grade;
        }

        public int getPoint()
        {
            return this.point;
        }

        public void setPoint(int point)
        {
            this.point = point;
        }

    }

}